.circleci/config.yml

```diff
@@ -4,17 +4,103 @@ executors:
     environment:
       ICR_IMAGE_NAME: icr.io/ext/logdna-agent
       IMAGE_NAME: logdna/logdna-agent
+      MAJOR_VERSION: 1.
     docker:
       - image: circleci/buildpack-deps:stretch
+  linux-publisher:
+    environment:
+      DEBIAN: deb
+      DESCRIPTION: LogDNA Agent for Linux
+      INPUT_TYPE: dir
+      LICENSE: MIT
+      MAINTAINER: support@logdna.com
+      MAJOR_VERSION: 1.
+      NAME: logdna-agent
+      REDHAT: rpm
+      URL: http://logdna.com/
+      VENDOR: LogDNA, Inc.
+    docker:
+      - image: circleci/node:8.3.0
+      - image: circleci/ruby
 jobs:
   test:
     docker:
-      - image: circleci/node:10
+      - image: circleci/node:8.3.0
     steps:
       - checkout
       - run: npm install
       - run: npm run lint
       - run: npm run test
+  linux-build:
+    executor: linux-publisher
+    steps:
+      - checkout
+      - run:
+          name: Check Tagged Version
+          command: |
+            TAG_VERSION=$(git tag --sort=-creatordate | head -n 1)
+            PKG_VERSION=$(cat package.json | grep version | cut -d':' -f2 | cut -d '"' -f2)
+            if [[ "${TAG_VERSION}" != "${PKG_VERSION}" || "${TAG_VERSION}" != "${CIRCLE_TAG}" || ${CIRCLE_TAG} != ${MAJOR_VERSION}* ]]; then
+              echo "There is mismatch:"
+              echo "  TAG_VERSION: ${TAG_VERSION}"
+              echo "  PKG_VERSION: ${PKG_VERSION}"
+              exit 1
+            fi
+            echo "export VERSION=${CIRCLE_TAG}" >> ${BASH_ENV}
+            source ${BASH_ENV}
+      - run:
+          name: Install Dependencies
+          command: |
+            npm install -g nexe
+            gem install --no-ri --no-rdoc fpm
+            mkdir -p .build/scripts
+            cp ./scripts/linux/* .build/scripts/
+      - run:
+          name: Build Executable
+          command: nexe -i index.js -o .build/logdna-agent -t linux-x64-8.3.0
+      - run:
+          name: Package for Debian
+          command: |
+            fpm \
+              --input-type ${INPUT_TYPE}\
+              --output-type ${DEBIAN}\
+              --name ${NAME}\
+              --version ${VERSION}\
+              --license ${LICENSE}\
+              --vendor ${VENDOR}\
+              --description ${DESCRIPTION}\
+              --url ${URL}\
+              --maintainer ${MAINTAINER}\
+              --before-remove ./.build/scripts/before-remove\
+              --after-upgrade ./.build/scripts/after-upgrade\
+              --force --deb-no-default-config-files\
+                ./.build/logdna-agent=/usr/bin/logdna-agent\
+                ./.build/scripts/init-script=/etc/init.d/logdna-agent\
```

Comment:
We don't ship a SystemD unit file?

Reply:
no - idk why but it's been like this for always

Reply:
I'll dig around and see if this is something we can add, I guess Centos6 still needs a SystemV file so. Anyways. 👍